### ISSUE TITLE
show a message if firebase fails to connect

### ISSUE THREAD

**Issue by bqi343** [cpinitiative/usaco-guide#3573]
maybe we should show a message if firebase fails to connect…

_Originally posted by @thecodingwizard in https://github.com/cpinitiative/usaco-guide/issues/3425#issuecomment-1472360523_

**Comment by Javeed9** [comment]
Hey, can i resolve this issue?

**Comment by bqi343** [comment]
sure

**Comment by bqi343** [comment]
> @mjh316 pushed something that might be relevant in https://github.com/cpinitiative/usaco-guide/pull/3594 . Enabling longpolling sometimes fixes this issue for users with stable internet connections but firebase connection issues

@mjh316 @thecodingwizard can we add a message to https://usaco.guide/settings next to long polling that mentions this?

**Comment by mjh316** [comment]
> > @mjh316 pushed something that might be relevant in https://github.com/cpinitiative/usaco-guide/pull/3594 . Enabling longpolling sometimes fixes this issue for users with stable internet connections but firebase connection issues
> 
> 
> 
> @mjh316 @thecodingwizard can we add a message to https://usaco.guide/settings next to long polling that mentions this?

Yes, I put an initial message there as I wasn't sure what to say - do you have a specific message in mind we could use?. Also, is there a way we can hook into the connection failure and tell them to attempt long polling?

**Comment by bqi343** [comment]
@mjh316 Add something about enabling this option only if you encounter issues connecting to Firebase.

**Comment by alstondsouza1** [comment]
Hi! I'm a student in the CodePath AI301 Open Source Capstone program and I'd like to work on this issue.

From reading the discussion, I understand there was a previous change related to long polling and Firebase connection issues, but the issue remains open. I'll start by reproducing the behavior locally and reviewing the existing implementation.

Are there any specific areas of the codebase I should look at first, or any remaining requirements that are not yet addressed?

Thank you!

**Comment by alstondsouza1** [comment]
I set up the project locally and found that `src/components/Settings/General.tsx` already includes a long polling message: “Enable this option ONLY if you encounter issues connecting to Firebase (nothing loads).”

Since this appears to match the earlier maintainer suggestion, should this issue be considered resolved, or is there still additional behavior expected, such as detecting Firebase connection failure and showing a dynamic message?

**Comment by bqi343** [comment]
yes I think it's fine now
